diff --git a/src/feature/ALIKED.cpp b/src/feature/ALIKED.cpp
index 5aae9df..1101e7e 100644
--- a/src/feature/ALIKED.cpp
+++ b/src/feature/ALIKED.cpp
@@ -111,7 +111,7 @@ ALIKED::forward(torch::Tensor image) && {
     auto [descriptors, offsets] = std::move(*desc_head_).forward(feature_map, keypoints);
 
     auto end_time = std::chrono::high_resolution_clock::now();
-    auto duration = duration_cast<std::chrono::milliseconds>(end_time - start_time).count() / 1000.0f;
+    auto duration = std::chrono::duration_cast<std::chrono::milliseconds>(end_time - start_time).count() / 1000.0f;
 
     torch::Dict<std::string, torch::Tensor> output;
     output.insert("keypoints", std::move(keypoints[0]));
diff --git a/src/feature/get_patches_cuda.cu b/src/feature/get_patches_cuda.cu
index 32e1be2..9285d95 100644
--- a/src/feature/get_patches_cuda.cu
+++ b/src/feature/get_patches_cuda.cu
@@ -9,7 +9,7 @@
 
 namespace F = torch::nn::functional;
 
-#define CHECK_CUDA(x)       TORCH_CHECK(x.type().is_cuda(), #x " must be a CUDA tensor")
+#define CHECK_CUDA(x)       TORCH_CHECK(x.is_cuda(), #x " must be a CUDA tensor")
 #define CHECK_CONTIGUOUS(x) TORCH_CHECK(x.is_contiguous(), #x " must be contiguous")
 #define CHECK_INPUT(x) \
     CHECK_CUDA(x);     \
@@ -159,7 +159,7 @@ torch::Tensor get_patches_forward_cuda(const torch::Tensor& input, torch::Tensor
     // cuda kernel
     int64_t num_kernels = n_input_plane * n_points;
     auto stream = at::cuda::getCurrentCUDAStream();
-    AT_DISPATCH_FLOATING_TYPES(input.type(), "get_patches_forward_cuda",
+    AT_DISPATCH_FLOATING_TYPES(input.scalar_type(), "get_patches_forward_cuda",
                                (
                                    [&] {
                                        get_patches_forward_cuda_kernel<scalar_t><<<GET_BLOCKS(num_kernels), CUDA_NUM_THREADS, 0, stream>>>(
@@ -196,7 +196,7 @@ torch::Tensor get_patches_forward_cuda1(const torch::Tensor& map, torch::Tensor&
     // cuda kernel
     const int threads = CUDA_NUM_THREADS;
     const int blocks = (N + threads - 1) / threads;
-    AT_DISPATCH_FLOATING_TYPES(map_pad.type(), "get_patches_forward_cuda",
+    AT_DISPATCH_FLOATING_TYPES(map_pad.scalar_type(), "get_patches_forward_cuda",
                                (
                                    [&] {
                                        get_patches_forward_cuda_kernel1<scalar_t>
@@ -234,7 +234,7 @@ torch::Tensor get_patches_backward_cuda(const torch::Tensor& d_patches, torch::T
     // cuda kernel
     const int threads = CUDA_NUM_THREADS;
     const int blocks = (N + threads - 1) / threads;
-    AT_DISPATCH_FLOATING_TYPES(d_map_pad.type(), "get_patches_backward_cuda",
+    AT_DISPATCH_FLOATING_TYPES(d_map_pad.scalar_type(), "get_patches_backward_cuda",
                                (
                                    [&] {
                                        get_patches_backward_cuda_kernel<scalar_t>
